### ISSUE TITLE
Consideration of map vision limit on token vision

### ISSUE THREAD

**Issue by Tixuoty66** [encounterplus/encounterplus#446]
My understanding of the current way of working : 

When a map vision limit is set, token vision will be the smallest value between ' token vision' and 'map vision limit'. So a token with a 60' darkvision and a map vision limit of 40' will have a 40' los. But with a map vision limit of 80', the same token will be limited to his 60' darkvision even in full daylight.

Suggestion : if 'daylight' map setting > 0 and token vision < map vision limit, then token vision should be map vision limit one.

**Comment by Splinter714** [comment]
I'm not sure I fully understand the suggestion right at the end. Everything made sense until the last sentence. Could you elaborate a bit, please?

**Comment by Tixuoty66** [comment]
I'll try: if daylight > 0 (example in full daylight), token vision seems not correct when map vision limit is higher than that of token vision. In daylight, a token with a 60' darkvision should see up to map vision limit if this one is of 80' for example. Which is not the case with current release where token vision will be limited to 60'. The 'suggestion' is my thought of the way it could be calculated.

**Comment by Splinter714** [comment]
Ah, yes that makes perfect sense. Thank you for explaining further.

**Comment by Rycochet** [comment]
(To put a bit of real-world context - imagine it's a foggy daytime, even though it's light out, you still can't see very far, but you can see further than at night)

**Comment by Tixuoty66** [comment]
Exactly, that’s why your token visibility is limited to the map vision set and not beyond.

**Comment by jurex** [comment]
I need to change the implementation for this, especially disable daylight enforcement.
